---
title: "Untitled"
author: "Leonard Sparring"
date: "5/21/2019"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(DESeq2)
```



```{r, include=TRUE}

sampleCondition <- c('BH','BH','BH','Serum','Serum','Serum')

sampleFiles <- grep("csv",list.files("/home/leo/Documents/GenomeAnalysis/GenomeAnalysis/genome_analysis/analyses/190516_htseq_count_data/"),value=TRUE)

sampleTable <- data.frame(sampleName = c('BH_1','BH_2','BH_3','Serum_1','Serum_2','Serum_3'),
                          fileName = sampleFiles,
                          condition = sampleCondition)

ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                                       directory = "/home/leo/Documents/GenomeAnalysis/GenomeAnalysis/genome_analysis/analyses/190516_htseq_count_data",
                                       design= ~ condition)
ddsHTSeq

```

```{r, include=TRUE}
# Filtering out the hypothetical proteins
mykeep <- !grepl("KAEIAEFF*", rownames(counts(ddsHTSeq)))
names(mykeep) <- rownames(counts(ddsHTSeq))
nrow(ddsHTSeq)
ddsHTSeq <- ddsHTSeq[mykeep,]
nrow(ddsHTSeq)
counts(ddsHTSeq)

#Pre-filtering
keep <- rowSums(counts(ddsHTSeq)) >= 10
ddsHTSeq <- ddsHTSeq[keep,]
counts(ddsHTSeq)
nrow(ddsHTSeq)


```
```{r, include=TRUE}

# Differential expression analysis

ddsHTSeq <- DESeq(ddsHTSeq)
res <- results(ddsHTSeq)
res
```

```{r, include=TRUE}
# Ordered p-values

resOrdered <- res[order(res$pvalue),]

summary(res)

sum(res$padj < 0.1, na.rm=TRUE) # Way too many?




```

```{r, include=TRUE}

# Exporting most significant genes to csv

resSig <- subset(resOrdered, padj < 0.01)
resSig

write.csv(as.data.frame(resOrdered), 
          file="condition_Serum_vs_BH_pval0point01.csv")

```


```{r, include=TRUE}
# Log fold change shrinkage for visualization and ranking

resultsNames(ddsHTSeq)

resLFC <- lfcShrink(ddsHTSeq, coef="condition_Serum_vs_BH", type="apeglm")
resLFC

```

```{r, include=TRUE}

# Plot of shrunken log2 fold changes
plotMA(resLFC, ylim=c(-2,2))

```


```{r, include=TRUE}
# Heatmap

ntd <- normTransform(ddsHTSeq)
print(ntd)

library("pheatmap")
select <- order(rowMeans(counts(ddsHTSeq,normalized=TRUE)),
                decreasing=TRUE)[1:20]
df <- as.data.frame(colData(ddsHTSeq)[,c("condition","sizeFactor")])
pheatmap(assay(ntd)[select,], cluster_rows=FALSE, show_rownames=FALSE,
         cluster_cols=FALSE, annotation_col=df)
```


