---
title: "Differential Expression Analysis"
author: "Leonard Sparring"
date: "5/21/2019"
output:
  html_document:
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(DESeq2)
```



```{r, include=TRUE}

sampleCondition <- c('BH','BH','BH','Serum','Serum','Serum')

sampleFiles <- grep("csv",list.files("/home/leo/Documents/GenomeAnalysis/GenomeAnalysis/genome_analysis/analyses/190516_htseq_count_data/"),value=TRUE)

sampleTable <- data.frame(sampleName = c('BH_1','BH_2','BH_3','Serum_1','Serum_2','Serum_3'),
                          fileName = sampleFiles,
                          condition = sampleCondition)

ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                                       directory = "/home/leo/Documents/GenomeAnalysis/GenomeAnalysis/genome_analysis/analyses/190516_htseq_count_data",
                                       design= ~ condition)
ddsHTSeq

```

```{r, include=TRUE}
# Filtering out the hypothetical proteins
#mykeep <- !grepl("KAEIAEFF*", rownames(counts(ddsHTSeq)))
#names(mykeep) <- rownames(counts(ddsHTSeq))
#nrow(ddsHTSeq)
#ddsHTSeq <- ddsHTSeq[mykeep,]
#nrow(ddsHTSeq)
#counts(ddsHTSeq)

#Pre-filtering
keep <- rowSums(counts(ddsHTSeq)) >= 10
ddsHTSeq <- ddsHTSeq[keep,]
counts(ddsHTSeq)
nrow(ddsHTSeq)


```
```{r, include=TRUE}

# Differential expression analysis

ddsHTSeq <- DESeq(ddsHTSeq)
res <- results(ddsHTSeq)
res
```

```{r, include=TRUE}
# Ordered genes by log2foldchanges

resOrdered <- res[order(res$log2FoldChange),]

summary(res)
res05 <- results(ddsHTSeq, alpha=0.05)
summary(res05)


sum(res$padj < 0.1, na.rm=TRUE) # Way too many?




```

```{r, include=TRUE}

# Exporting most significant genes to csv

resSig <- subset(resOrdered, padj < 0.01)
resSig

write.csv(as.data.frame(resSig), 
          file="condition_Serum_vs_BH.csv")

write.csv(as.data.frame(resSig[ order( -resSig$log2FoldChange, -resSig$baseMean ),]), file="UpRegulated.csv")

```


```{r, include=TRUE}
# Log fold change shrinkage for visualization and ranking

resultsNames(ddsHTSeq)

resLFC <- lfcShrink(ddsHTSeq, coef="condition_Serum_vs_BH", type="apeglm")
resLFC

```

```{r, include=TRUE}

# Plot of shrunken log2 fold changes
plotMA(resLFC, ylim=c(-2,2))

```

# Heatmap of downregulated differentially expressed genes. normalized counts ordered by most differentially log fold changes

```{r, include=TRUE}

library("pheatmap")

ntd <- normTransform(ddsHTSeq)

select <- order(rowMeans(counts(ddsHTSeq,normalized=TRUE)),
                decreasing=TRUE)

rowMeansCounts <- assay(ntd)[select,]
topVarGenes <- subset(rowMeansCounts, rownames(rowMeansCounts) %in% row.names(resOrdered)[1:20])


df <- as.data.frame(colData(ddsHTSeq)[,c("condition","sizeFactor")])
pheatmap(topVarGenes, cluster_rows=FALSE, show_rownames=TRUE,
         cluster_cols=FALSE, annotation_col=df)

```

# DE of upregulated genes

```{r, include=TRUE}


select <- order(rowMeans(counts(ddsHTSeq,normalized=TRUE)),
                decreasing=TRUE)

resNegOrdered <- res[order(-res$log2FoldChange),]

rowMeansCounts <- assay(ntd)[select,]
topVarNegGenes <- subset(rowMeansCounts, rownames(rowMeansCounts) %in% row.names(resNegOrdered)[1:20])


df2 <- as.data.frame(colData(ddsHTSeq)[,c("condition","sizeFactor")])
pheatmap(topVarNegGenes, cluster_rows=FALSE, show_rownames=TRUE,
         cluster_cols=FALSE, annotation_col=df2)





```


# Ordered by p-value

```{r, include=TRUE}


select <- order(rowMeans(counts(ddsHTSeq,normalized=TRUE)),
                decreasing=TRUE)

resPvalOrdered <- res[order(res$pvalue),]

rowMeansCounts <- assay(ntd)[select,]
topVarPvalGenes <- subset(rowMeansCounts, rownames(rowMeansCounts) %in% row.names(resPvalOrdered)[1:20])


df3 <- as.data.frame(colData(ddsHTSeq)[,c("condition","sizeFactor")])
pheatmap(topVarPvalGenes, cluster_rows=FALSE, show_rownames=TRUE,
         cluster_cols=FALSE, annotation_col=df3)




```


